### DISCUSSION TITLE
Demo: LayoutStrategy with horizontal/vertical placement of children and support for percentage-based width/height

### DISCUSSION THREAD

**Discussion by drinkspiller** [fabricjs/fabric.js#10127] [Q&A]
I put together a demo of "inline" LayoutStrategy where the children of a group are positioned either horizontally (row) or vertically (column). The demo also supports supplying percentage-based width and height for Group and Shapes.

[DEMO](https://codesandbox.io/p/devbox/xyvyt6?file=%2Fsrc%2Ffabric_layout_strategy.ts%3A9%2C1)

I've read the [LayoutManager README](https://github.com/fabricjs/fabric.js/blob/master/src/LayoutManager/README.md) but am still not quite clear on when it is appropriate to implement layout logic in `LayoutStrategy`'s `calcLayoutResult` method vs `LayoutManager`'s `performLayout` method. I'd be grateful for a look at the `LayoutStrategy` implemented in the demo and suggestions for patterns that might be more idiomatic to Fabric. Thanks!

- Mr. Furley

**Comment by drinkspiller** [comment]
Friendly bump requesting guidance as to whether this should be done in `LayoutStrategy`'s `calcLayoutResult` method vs `LayoutManager`'s `performLayout` method?

**Reply by asturur** [reply]
Sorry for not replying.
The readme was my attempt to document how the layout strategy should have worked from the contributor  code.
I cannot open your codesanbox demo anymore, or i would happily read it.
Can you make sure the demo is reacheable?

**Comment by graphics-et-al** [comment]
@drinkspiller Interested in this too- would love to see the demo

**Reply by drinkspiller** [reply]
@asturur & @graphics-et-al 

It kind of works, but it's brittle. All the objects have to be added to the top level group followed by an imperative call to `topLevelGroup.triggerLayout()`. I ran into issues trying to 1) remove the need to imperatively call `triggerLayout()` and 2) getting layout working correctly if objects are added after the initial layout.

To see where things stand:
1. Try this [updated demo link](https://codesandbox.io/p/devbox/dreamy-zeh-forked-yq6vk7?workspaceId=932daf75-7b4c-4c6a-bc71-03b66ca80529)
2. Start the preview:
![image](https://github.com/user-attachments/assets/6e280c11-5b8c-440f-bd4d-033c9834b8d7)
3. Check out: `src/fabric_layout_flex_strategy.ts`, `src/app/app.component.ts`

Feedback/modifications on 1) removing the requirement to imperatively `triggerLayout()` and 2) support layout for items `.add()`ed after the fact welcome!
